RealmSwift/Sync.swift

```diff
@@ -469,6 +469,97 @@ fileprivate extension SyncManagementObjectStatus {
     }
 }
 
+public extension SyncSession {
+    /// Whether a given progress notification block will report the upload status or the download status of its session.
+    public enum ProgressDirection {
+        /// :nodoc:
+        case upload
+        /// :nodoc:
+        case download
+    }
+    /// How the progress notification block will behave.
+    public enum ProgressMode {
+        /**
+         The block will be called forever, or until the user unregisters it.
+         The progress struct will always report the latest number of transferred bytes,
+         and the latest number of total bytes to be transferred.
+         */
+        case reportIndefinitely
+        /**
+         The block will, upon registration, store the total number of bytes
+         to be transferred. The progress struct will always report the latest number
+         of transferred bytes out of that original number of bytes to be
+         transferred. When the number of transferred bytes reaches or exceeds the
+         number of bytes to be transferred, the block will be automatically
+         unregistered.
+         */
+        case forCurrentlyOutstandingWork
+    }
+
+    /**
+     A token corresponding to a progress notification block.
+
+     Call `stop()` on the token to stop notifications. If the block has already
+     been automatically stopped, calling `stop()` does nothing. `stop()` should be called
+     before the token is destroyed.
+     */
+    public typealias NotificationToken = RLMProgressNotificationToken
+
+    /**
+     A struct encapsulating progress information, as well as useful helper methods.
+     */
+    public struct Progress {
+        /// The number of bytes that have been transferred.
+        public let bytesTransferred: UInt
+        /// The total number of bytes that have been transferred, and are pending transfer.
```

Comment:
I'm a bit confused by this `bytesPending` property. I'd expect `bytesPending` to decrease as data as uploaded, but it looks from how it's used within `fractionTransferred` that this value does not change as data is uploaded. Am I missing something?